### ISSUE TITLE
Repetition count in schedule string should be decremented by 1

### ISSUE THREAD

**Issue by MarcoVigna17** [ajvb/kala#112]
Looking on your readme, on che schedule string section, is written that `R1 - Will repeat once`.

So I was expecting just one run on the job that I scheduled with R1 repetition, but it happened twice. Then I investigated and I noticed that basically every job that was scheduled with a fixed repetition count has been executed 1 time more than expected.

How to reproduce the problem:
Those are two examples that I run to figure out the problem:
1 - I set up a job to run with 1 repetition
```
curl http://127.0.0.1:8000/api/v1/job/ -d '{"command": "bash echo A >> ~/Desktop/tmp_log_kala", "name": "test_job2", "schedule": "R1/2016-05-03T15:00:00+01:00/PT10S"}'
```
and this is the response from kala: `{"id":"8affc3e9-2837-415f-4aa0-0ef601611b16"}`
But looking at the GET job API after the end of the execution, I found two execution of it:
```
{
   "job" : {
      "next_run_at" : "2016-05-03T15:00:10.002875991+01:00",
      "parent_jobs" : null,
      "retries" : 0,
      "schedule" : "R1/2016-05-03T15:00:00+01:00/PT10S",
      "epsilon" : "",
      "metadata" : {
         "last_error" : "2016-05-03T15:00:10.016369805+01:00",
         "last_success" : "0001-01-01T00:00:00Z",
         "error_count" : 2,
         "success_count" : 0,
         "last_attempted_run" : "2016-05-03T15:00:10.008073338+01:00"
      },
      "disabled" : false,
      "id" : "8affc3e9-2837-415f-4aa0-0ef601611b16",
      "owner" : "",
      "command" : "bash echo A >> ~/Desktop/tmp_log_kala",
      "name" : "test_job2",
      "stats" : [
         {
            "number_of_retries" : 0,
            "job_id" : "8affc3e9-2837-415f-4aa0-0ef601611b16",
            "ran_at" : "2016-05-03T15:00:00.002876963+01:00",
            "execution_duration" : 11777780,
            "success" : false
         },
         {
            "success" : false,
            "job_id" : "8affc3e9-2837-415f-4aa0-0ef601611b16",
            "execution_duration" : 8295465,
            "ran_at" : "2016-05-03T15:00:10.00807454+01:00",
            "number_of_retries" : 0
         }
      ],
      "dependent_jobs" : null
   }
}
```
2 - Then I tried forcing a 0 repetition for my job (that in theory has no sense)
```
curl http://127.0.0.1:8000/api/v1/job/ -d '{"command": "bash echo A >> ~/Desktop/tmp_log_kala", "name": "test_job2", "schedule": "R0/2016-05-03T15:00:00+01:00/PT10S"}'
```
and the response was a job created correctly `{"id":"f44e86bf-feb3-4e4d-5580-d9850583d623"}`
But looking at the execution I found that was executed once:
```
{
   "job" : {
      "owner" : "",
      "metadata" : {
         "last_success" : "0001-01-01T00:00:00Z",
         "last_attempted_run" : "2016-05-03T15:00:00.002873922+01:00",
         "success_count" : 0,
         "error_count" : 1,
         "last_error" : "2016-05-03T15:00:00.013881925+01:00"
      },
      "schedule" : "R0/2016-05-03T15:00:00+01:00/PT10S",
      "parent_jobs" : null,
      "retries" : 0,
      "id" : "f44e86bf-feb3-4e4d-5580-d9850583d623",
      "next_run_at" : "2016-05-03T15:00:00.00000066+01:00",
      "stats" : [
         {
            "ran_at" : "2016-05-03T15:00:00.002874753+01:00",
            "number_of_retries" : 0,
            "execution_duration" : 11007273,
            "job_id" : "f44e86bf-feb3-4e4d-5580-d9850583d623",
            "success" : false
         }
      ],
      "command" : "bash echo A >> ~/Desktop/tmp_log_kala",
      "epsilon" : "",
      "dependent_jobs" : null,
      "disabled" : false,
      "name" : "test_job2"
   }
}
```

Is the documentation that is wrong or is kala itself?
At the moment I'm forcing my jobs to be repeated n-1 times, but I'd like to know if I have to revert this hack in the future or not...

Regards, 
Marco

**Comment by ajvb** [comment]
@MarcoVigna17 - Neither, but I understand how it could seem that way. I will try to make the docs more clear.

It's in the term `repeat`. Repeating once != Running once. The `R` value is how many times it will repeat after its initial run, as you would never have a Run Zero Times job.

**Comment by benwaine** [comment]
Can I ask for some clarification around the repeat interval: 

As per the docs the time string has three sections, repetition value, start date and interval. 

```
R/2016-05-05T15:00:00+01:00/PT5S
```

Should the interval be required when scheduling a job with an `R` value of 0? 

I ran some tests: 

```
// Success - R0 with a (redundant?) interval
baw$ curl http://127.0.0.1:8000/api/v1/job/ -d '{"command": "bash echo AA", "name": "test_job", "schedule": "R0/2016-05-05T15:00:00+01:00/PT10S"}'
{"id":"e9f8cc6c-9aa2-48d5-4d9e-046c5dc80c1f"}

// Failure - R0 with no interval
baw$ curl http://127.0.0.1:8000/api/v1/job/ -d '{"command": "bash echo AA", "name": "test_job", "schedule": "R0/2016-05-05T15:00:00+01:00/"}'
Error occured when initializing the job

// Failure - R0 with no interval
baw$ curl http://127.0.0.1:8000/api/v1/job/ -d '{"command": "bash echo AA", "name": "test_job", "schedule": "R0/2016-05-05T15:00:00+01:00"}'

```

**Comment by ajvb** [comment]
@benwaine I agree, the interval should absolutely not be required when you are specifying a job with a repeat value of zero.

**Comment by ajvb** [comment]
@benwaine Fixed by https://github.com/ajvb/kala/pull/116

**Comment by benwaine** [comment]
Thanks, I'll build master today and take a look.

**Comment by MarcoVigna17** [comment]
This has created the bug explained at #118 fixed by #119
